### ISSUE TITLE
Ability to commend fold

### ISSUE THREAD

**Issue by mitermayer** [scrooloose/nerdcommenter#315]
It would be awesome if we could comment a fold without having to visually select lines.

**Comment by alerque** [comment]
You probably can. Does vim not have motion or range commands for the current fold?

**Comment by mitermayer** [comment]
Would be nice if that was part of nerdcommenter

**Comment by alerque** [comment]
I'm unsure what you think this would look like. The way vim and plugins typically work is that each plugin handles some specific task and provides specific functions. Good plugins allow these functions to interact with other vim functions and the user can mix and match them as their workflow requires.

* NerdCommenter provides comment related functions.
* Vim  provides range, motion, selection, [text objects](https://blog.carbonfive.com/2011/10/17/vim-text-objects-the-definitive-guide/), etc.
* You can run any mode of `NERDComment()` on any of the above.

If this plugin were to try to get into pairing the above on it's own instead of leaving that to the user the possibilities would mushroom out of control really fast. The iterations of which comment style (line? block? sexy? minimal?) and motion type (word, sentence, paragraph, quoted string, fold, indent level, inside? outside?) would be in the thousands.

Instead the way vim works is you mix and match yourself. If a particular combination of motion and action is too cumbersome for you, you can make it more accessible with a mapping.

I assume you've read the `:help fold` documentation for how to move around and manipulate folds. If you want to operate on them a convenience text object for the current fold can be handy. For example of how to do this see [this tutorial](vim.wikia.com/wiki/Creating_new_text_objects).

The short version is you'll want these two lines in your rc file:

```vim
vnoremap af :<C-U>silent! normal! [zV]z<CR>
omap af :normal Vaf<CR>
```

Now you have text object `af` (a fold). Lets say you like the sexy comment style and want to have quick access to that comment style for the current fold:

```vim
map <leader>cf Vaf:call NERDComment(1, 'sexy')<Cr>
```
